e2e/async-esm-import/package.json

```diff
@@ -0,0 +1,13 @@
+{
+  "name": "async-esm-import",
+  "jest": {
+    "testEnvironment": "node",
+    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(js?|ts?|mjs?|mts?)$",
+    "moduleFileExtensions": [
+      "js",
+      "ts",
+      "mjs",
+      "mts"
+    ]
```

Comment:
These lines look redundant. They add complexity and this is not what is being tested.

What if the test file would have `.js`, but the imported file `.mjs` extension? That was my initial idea. This way the test would get transformed and executed as CJS (no need to import `jest`). The imported module would get interpreted as ESM. A bit of both worlds in one test.

Alternatively keep `.js` extension for both files; add `"type": "module"` to `package.json`; and add `transform: {}` to Jest config.

Reply:
If remove these config, Jest will produce next info:

 ```sh
 exitCode: 1,
signal: undefined,
signalDescription: undefined,
 stdout: 'No tests found, exiting with code 1\n' +
        'Run with `--passWithNoTests` to exit with code 0\n' +
        'In /Users/eryue0220/cin/eryue/jest/e2e/async-esm-import\n' +
        '  3 files checked.\n' +
        '  testMatch: **/__tests__/**/*.[jt]s?(x), **/?(*.)+(spec|test).[tj]s?(x) - 0 matches\n' +
        '  testPathIgnorePatterns: /node_modules/ - 3 matches\n' +
        '  testRegex:  - 0 matches\n' +
        'Pattern:  - 0 matches',
 stderr: '',
```
the default `testMatch` doesn't match `.mjs` files. But the `moduleFileExtensions` is really redundant.

I changed to the `.js` extensions.